src/main/java/com/example/music/backend/playlist/controller/PlaylistController.java

```diff
@@ -0,0 +1,73 @@
+package com.example.music.backend.playlist.controller;
+
+import com.example.music.backend.playlist.dto.PlaylistDto;
+import com.example.music.backend.playlist.response.PlaylistResponse;
+import com.example.music.backend.playlist.service.PlaylistService;
+import com.example.music.backend.user.domain.User;
+import com.example.music.backend.user.service.UserService;
+import com.example.music.backend.video.dto.YoutubeVideoDto;
+import com.example.music.backend.video.response.YoutubeVideoResponse;
+import lombok.AllArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.List;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@AllArgsConstructor
+@RestController
+@RequestMapping("/playlists")
+public class PlaylistController {
+
+    private final PlaylistService playlistService;
+    private final UserService userService;
+
+    @PutMapping(value = "/{playlistId}", produces = APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<YoutubeVideoResponse>> addVideoToPlaylist(
+            @PathVariable("playlistId") Long playlistId,
+            @RequestBody YoutubeVideoDto dto) {
+
+        playlistService.addVideo(playlistId, dto.getYoutubeId());
+        List<YoutubeVideoResponse> videos = playlistService.getAllVideoInPlaylist(playlistId);
+        return ResponseEntity.ok(videos);
+    }
+
+    @DeleteMapping(value = "/{playlistId}", produces = APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<YoutubeVideoResponse>> removeVideoFromPlaylist(
+            @PathVariable("playlistId") Long playlistId,
+            String youtubeId) {
+
+        playlistService.removeVideo(playlistId, youtubeId);
+        List<YoutubeVideoResponse> videos = playlistService.getAllVideoInPlaylist(playlistId);
+        return ResponseEntity.ok(videos);
+    }
+
+    @PostMapping(produces = APPLICATION_JSON_VALUE)
+    public ResponseEntity<PlaylistResponse> create(@RequestBody PlaylistDto playlistDto) {
+        Long userId = userService.getUserByEmail(
+                        SecurityContextHolder
+                                .getContext()
+                                .getAuthentication()
+                                .getName())
+                .getId();
```

Comment:
So dangerous. Possible place for NPE

Reply:
Why? If the user is not authorized then the exception will be Unauthorized, if the user is authorized then the ContextHolder will return an email, if the user is not found in the repository it will execute NotFoundExeption, where can NPE happen?

Reply:
Ok, you're right